### ISSUE TITLE
splitter?

### ISSUE THREAD

**Issue by jots** [chjj/blessed#36]
Has anyone implemented a splitter with blessed?  Something like the splitters that are in tmux that you can move around with your mouse (or keyboard)   and which resizes elements on either side of it.

**Comment by leshy** [comment]
hey, I have a similar idea but different aproach, 
why not use tmux itself?

I'm doing something like:

spawn = require('child_process').spawn
proc = spawn '/usr/bin/tmux', [ 'split-window', '-h', process.argv.join(" ") ]

which tells tmux to split the window and spawn one more of my app in this new window, I can then use some IPC method to talk to the "child"

node apps are slow to init though, (fork() might be faster, but I don't know how to delegate my new childs stdio to tmux then) and depending on how dependant these windows are on each other, IPC api might be large.

so I got another idea, but not sure if its entirely doable,

I could create a new screen object in blessed, which is actually a virtual screen representing my new tab/pane.

this screen object when initialized would spawn a tmux window/pane with a light STDIO proxy app within it, which talks to this screen object in my main app via a socket.

which makes my screens share the address space, and eliminates long initialization.
what do you think?

am I making sense? 
what do you think of this? I'll look at screen object code at some point and implement this if its not too ugly.

**Comment by chjj** [comment]
leshy, a better idea is to just use a Line element and then keep track of `move` events on it to determine the width and height of elements. That would be possible in probably just a few lines of code. With `dockBorders` the Line element will automatically dock to each border too.

**Comment by secrettriangle** [comment]
I think that a splitter is a great idea. :+1:

**Comment by chjj** [comment]
We can implement it as a child of `Line` with a few events bound to it and some options.

**Comment by leshy** [comment]
Yeah, I know this is super easily doable from within blessed, but my motives are more philosophical then pragmatic. I didn't like the idea of perfectly copying a window manager (tabs and splitting) if I'm already running within one. 

Also, when running within tmux I'd have two layers of keybindings for same actions but on different layers. 

I felt that this kind of tmux integration is a pretty cool aproach in general to "multi output" terminal apps, (ideally we don't want each app implementing its own window manager)
Just, I'm not very fond of my concrete implementation idea.

If I end up playing with it, I'll show you what I made, in case you care. The STDIO proxy thing might end up being some kind of a quick line in bash or socat, not sure.

**Comment by piranna** [comment]
Shouldn't this be easier to do with the overlaping boxes attaching the click and move events to them?
